src/loaders/BlueprintLoader.java

```diff
@@ -6,64 +6,93 @@
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.LinkedList;
+
+import com.sun.xml.internal.txw2.IllegalAnnotationException;
 
 public class BlueprintLoader {
-	
-	public static ArrayList<String> loadFromFile(String filePath){
+
+	public static ArrayList<String> loadFromFile(String filePath) {
 		ArrayList<String> processedLines = new ArrayList<>();
-		try(BufferedReader br = new BufferedReader(new FileReader(filePath))){
-			for(String line; (line = br.readLine()) != null;){
-				if(line.startsWith(">c")){
+		try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
+			for (String line; (line = br.readLine()) != null;) {
+				if (line.startsWith(">c")) {
 					continue;
-				}else{
-					processedLines.add(line);
+				} else {
+					processedLines.add(line.trim());
 				}
 			}
-		}catch (FileNotFoundException x){
+		} catch (FileNotFoundException x) {
 			x.printStackTrace();
-		}catch (IOException x){
+		} catch (IOException x) {
 			x.printStackTrace();
 		}
 		return processedLines;
 	}
-	
+
 	public static ArrayList<String> splitIntoLines(String blueprint) {
 		ArrayList<String> processedLines = new ArrayList<>();
 		processedLines.addAll(Arrays.asList(blueprint.split("\\n")));
 		return processedLines;
-		
+
 	}
-	
-	public static int extractEID(String input){
-		String substring = input.substring(
-				input.indexOf("eID=") + 4, 
-				input.indexOf("}"));
-		
-		return Integer.valueOf(substring);
+
+	public static int extractEID(String input) {
+		int idBeg = input.indexOf("eID=") + "eID=".length();
+		int idEnd = input.indexOf("}");
+
+		if (idBeg > 0) {
+			String subStr;
+			if (idEnd > idBeg) {
```

Comment:
I think this is supposed to check whether the structure COMPONENTNAME{eID=x}{} was followed and there are no other curly brace enclosed segments before the eID segment If I'm right on that, it should be made more clear. Also use the chance to give a warning about malformed input if you're testing for it anyways.